model/oauth/client.go

```diff
@@ -357,6 +368,14 @@ func (c *Client) Create(i *instance.Instance) *ClientRegistrationError {
 	c.GrantTypes = []string{"authorization_code", "refresh_token"}
 	c.ResponseTypes = []string{"code"}
 
+	// Adding Metadata
+	appSlug := c.SoftwareID
+	if strings.HasPrefix(c.SoftwareID, "registry://") {
+		appSlug = strings.TrimPrefix(c.SoftwareID, "registry://")
+	}
+	md, _ := metadata.NewWithApp(appSlug, c.SoftwareVersion, DocTypeVersion)
```

Comment:
> createdByApp: Slug of the app or connector which created the document

As it is described, I don't think we should put the `SoftwareID` (when it is not linked to a webapp) in the `createdByApp`. For example, we are using OAuth clients for sharing, and the SoftwareID in this case is `github.com/cozy/cozy-stack`.

cc: @edas

Reply:
The purpose is to know where the change or creation comes from. I don't mind having a "cozy-stack" slug, or even a "cozy/*" namespace for the stack. 
"github.com/cozy/cozy-stack" is somewhat out of scope IMHO.

If it's simplier for you, I agree to discard anything which is not on registry.

Also, be carefull not to include any channel in the slug (could a "registry://" string contain one ?)